activerecord/lib/active_record/relation/batches.rb

```diff
@@ -206,80 +212,51 @@ def find_in_batches(start: nil, finish: nil, batch_size: 1000, error_on_ignore:
     #
     # NOTE: By its nature, batch processing is subject to race conditions if
     # other processes are modifying the database.
-    def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore: nil, order: :asc, use_ranges: nil)
-      relation = self
-
-      unless [:asc, :desc].include?(order)
-        raise ArgumentError, ":order must be :asc or :desc, got #{order.inspect}"
+    def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore: nil, order: :asc, use_ranges: nil, &block)
+      if use_ranges == true && !VALID_DIRECTIONS.include?(order)
+        raise ArgumentError, ":order must be :asc or :desc when setting :use_ranges, got #{order.inspect}"
       end
 
-      unless block_given?
-        return BatchEnumerator.new(of: of, start: start, finish: finish, relation: self, order: order, use_ranges: use_ranges)
-      end
+      order = process_batches_order_arg(order)
```

Comment:
This changes what we pass to BatchEnumerator#initialize - the order fields are now an `Array` of `Arel::Nodes::Ordering`, rather than a Symbol. I don't know if this is considered to be a change to a public API?